### DISCUSSION TITLE
Reinventing

### DISCUSSION THREAD

**Discussion by smotched** [webstudio-is/webstudio-community#6] [General]
For the builder, would it be better to use or build on top of something like https://github.com/artf/grapesjs 

or is a new approach from the group up necessary?

**Comment by kof** [comment]
I am afraid its not that easy to build on top of something like grapesjs and it will be limiting in how we build it

**Reply by deckchairlabs** [reply]
I've had success with craftjs in previous projects https://github.com/prevwong/craft.js

**Reply by kof** [reply]
Nice one, won't work though. We want to achieve a certain UX and a tool like this will dictate it. In addition to that I want to add ability to have multiple pages with different breakpoints rendered on the canvas, that's why canvas is in the iframe, loaded as a standalone document

**Reply by kof** [reply]
On top of that there might be problems with real-time collaboration. Webstudio's architecture is designed with this in mind. Atomic updates for everything on the canvas.

**Reply by smotched** [reply]
@kof I was in favor of starting with a clean slate, but I wanted to make sure you thought about it first and we're not reinventing things just for the sake of it. tbh I still wasn't sure with your first comment, but these are good reasons, thanks for clarifying
